sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/SentrySpanProcessor.java

```diff
@@ -1,19 +1,95 @@
 package io.sentry.opentelemetry;
 
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
 import io.opentelemetry.context.Context;
 import io.opentelemetry.sdk.trace.ReadWriteSpan;
 import io.opentelemetry.sdk.trace.ReadableSpan;
 import io.opentelemetry.sdk.trace.SpanProcessor;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import io.sentry.DateUtils;
+import io.sentry.DsnUtil;
+import io.sentry.ISpan;
+import io.sentry.ITransaction;
+import io.sentry.Instrumenter;
 import io.sentry.Sentry;
+import io.sentry.SpanId;
+import io.sentry.SpanStatus;
+import io.sentry.TransactionContext;
+import io.sentry.TransactionOptions;
+import io.sentry.protocol.SentryId;
+import io.sentry.protocol.TransactionNameSource;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
-@SuppressWarnings("CatchAndPrintStackTrace")
 public final class SentrySpanProcessor implements SpanProcessor {
 
+  private final @NotNull Map<String, ISpan> spans = new ConcurrentHashMap<>();
+  private final @NotNull List<SpanKind> spanKindsConsideredForSentryRequests =
+      Arrays.asList(SpanKind.CLIENT, SpanKind.INTERNAL);
+  private final @NotNull SpanDescriptionExtractor spanDescriptionExtractor =
+      new SpanDescriptionExtractor();
+
   @Override
-  public void onStart(Context parentContext, ReadWriteSpan span) {
-    System.out.println(
-        "hello from onStart " + Thread.currentThread().getId() + Sentry.getCurrentHub().toString());
-    // TODO start
+  public void onStart(final @NotNull Context parentContext, final @NotNull ReadWriteSpan otelSpan) {
+    if (!hasSentryBeenInitialized()) {
+      return;
+    }
+
+    if (!Instrumenter.OTEL.equals(Sentry.getCurrentHub().getOptions().getInstrumenter())) {
+      return;
+    }
+
+    final @NotNull SpanContext otelSpanContext = otelSpan.getSpanContext();
+    if (!otelSpanContext.isValid()) {
+      return;
+    }
+
+    if (isSentryRequest(otelSpan)) {
```

Comment:
Just FYI, you may have considered this anyhow, but in the JS OTEL implementation, no attributes are set when `onStart` is called - attributes are (sadly) only available to check in `onEnd`, meaning we can't check like this. Just to keep this in mind, and make sure to test/try that these attributes are actually available here (in Ruby they are available, as far as I understood).

Reply:
@mydea I just double checked. Attributes are set in `onStart`. If the logic didn't work we'd be creating an infinite loop here as every request to Sentry would create another one - which is not the case.

I guess this means the JS SDK starts a new span for every request to Sentry that is then never finished. In Java this could be causing problems when combined with `waitForChildren` or `idleTimeout`. Also I guess we'd be spamming the UI with lots of unfinished spans in case they aren't filtered out. But I presume you've taken care of all those things or they're not relevant in JS.

Reply:
Ah no infinite loops (as that's prevented by the check in `onEnd`) just unnecessary Spans that might be causing the issues I just mentioned.

Reply:
yeah, exactly, unfortunately that's what we have to do in the JS SDK. They are never sent to the UI in that case, but "dropped" once we hit `onEnd`. It's obv. not ideal, but sadly no other way to handle this (*that we've found so far, at least).

but perfect if it works here 👍 just wanted to point this out as a learning we made the hard way ;)

Reply:
🙏 only thing i can think of atm is to only start the transaction in `onEnd` if we care about it (and set timestamps correctly) and totally ignore `onStart`

Reply:
The problem there is that you get into weird situation with parent spans etc, as when a span is finished you don't know about the parent yet (because that's not finished and thus wasn't handled yet). Anyways, lucky for you that seems to be a JS-only problem 😂